### ISSUE TITLE
Deletion of a Base Image gives rise to error

### ISSUE THREAD

**Issue by lorenzopagliai** [edgehog-device-manager/edgehog#588]
Once an Image Base has been created, it cannot be deleted directly from the Edgehog portal using the delete button, and the error that results is shown in the image below. 
![Screenshot from 2024-09-04 11-47-36](https://github.com/user-attachments/assets/60d114b6-5326-4fb2-8dfa-63f8f066e828)

**Comment by rbino** [comment]
Can you post the backend logs after the DELETE? I'm suspecting this is related to #586

**Comment by lorenzopagliai** [comment]
This is the log from Edgehog Backend pod:
```
level=info ts=2024-09-04T10:43:11.441+00:00 msg="POST /tenants/edgehogtenant/api" module=Phoenix.Logger function=phoenix_endpoint_start/4 request_id=fd4b4004f6eb1352a4d02f9b3b0433dc
level=error ts=2024-09-04T10:43:11.455+00:00 msg="d7b0b2a4-75fe-48ad-a894-1ca72ae19039: Exception raised while resolving query.\n\n** (Ash.Error.Unknown) Unknown Error\n\n* %Postgrex.Error{message: nil, postgres: %{code: :foreign_key_violation, line: \"2621\", message: \"update or delete on table \\\"base_images\\\" violates foreign key constraint \\\"update_campaigns_base_image_id_fkey\\\" on table \\\"update_campaigns\\\"\", table: \"update_campaigns\", file: \"ri_triggers.c\", unknown: \"ERROR\", constraint: \"update_campaigns_base_image_id_fkey\", severity: \"ERROR\", detail: \"Key (id, tenant_id)=(13, 1) is still referenced from table \\\"update_campaigns\\\".\", schema: \"public\", pg_code: \"23503\", routine: \"ri_ReportViolation\"}, connection_id: 3727, query: nil}\n  (ecto_sql 3.11.3) lib/ecto/adapters/sql.ex:1054: Ecto.Adapters.SQL.raise_sql_call_error/1\n  (ecto_sql 3.11.3) lib/ecto/adapters/sql.ex:952: Ecto.Adapters.SQL.execute/6\n  (ash_postgres 2.1.1) lib/data_layer.ex:2781: AshPostgres.DataLayer.destroy/2\n  (ash 3.1.1) lib/ash/actions/destroy/destroy.ex:196: anonymous fn/3 in Ash.Actions.Destroy.commit/3\n  (ash 3.1.1) lib/ash/changeset/changeset.ex:3421: Ash.Changeset.run_around_actions/2\n  (ash 3.1.1) lib/ash/changeset/changeset.ex:2962: anonymous fn/3 in Ash.Changeset.with_hooks/3\n  (ecto_sql 3.11.3) lib/ecto/adapters/sql.ex:1358: anonymous fn/3 in Ecto.Adapters.SQL.checkout_or_transaction/4\n  (db_connection 2.7.0) lib/db_connection.ex:1756: DBConnection.run_transaction/4\n\n    (ecto_sql 3.11.3) lib/ecto/adapters/sql.ex:1054: Ecto.Adapters.SQL.raise_sql_call_error/1\n    (ecto_sql 3.11.3) lib/ecto/adapters/sql.ex:952: Ecto.Adapters.SQL.execute/6\n    (ash_postgres 2.1.1) lib/data_layer.ex:2781: AshPostgres.DataLayer.destroy/2\n    (ash 3.1.1) lib/ash/actions/destroy/destroy.ex:196: anonymous fn/3 in Ash.Actions.Destroy.commit/3\n    (ash 3.1.1) lib/ash/changeset/changeset.ex:3421: Ash.Changeset.run_around_actions/2\n    (ash 3.1.1) lib/ash/changeset/changeset.ex:2962: anonymous fn/3 in Ash.Changeset.with_hooks/3\n    (ecto_sql 3.11.3) lib/ecto/adapters/sql.ex:1358: anonymous fn/3 in Ecto.Adapters.SQL.checkout_or_transaction/4\n    (db_connection 2.7.0) lib/db_connection.ex:1756: DBConnection.run_transaction/4\n\n" module=AshGraphql.Graphql.Resolver function=log_exception/2 request_id=fd4b4004f6eb1352a4d02f9b3b0433dc
level=info ts=2024-09-04T10:43:11.455+00:00 msg="Sent 200 in 13ms" module=Phoenix.Logger function=phoenix_endpoint_stop/4 request_id=fd4b4004f6eb1352a4d02f9b3b0433dc
```

**Comment by rbino** [comment]
@lorenzopagliai the issue here was that Edgehog is not allowing a Base Image to be deleted since it's used in an Update Campaign. The problem was that the error message was not very clear, this was fixed in #591 that will be contained in the upcoming `0.9.0-rc.2` release

**Comment by lorenzopagliai** [comment]
@rbino so once an Update Campaign is created, the corresponding Update Channel and Base Image Collection can no longer be deleted, am I wrong? Do you plan, at least in the long term, to provide the possibility to delete these objects in this scenario as well? We strongly believe that this could be a very convenient feature.  
Also, a simple note on the error message, which we actually see changed in release `0.9.0-rc.2`. In my opinion, it is not clear enough and should better emphasise what you have explained within this issue.

![Screenshot from 2024-09-12 10-58-55](https://github.com/user-attachments/assets/81132a17-bc3c-4a21-8b31-75dc2e218380)
